auth/plain.go

```diff
@@ -11,7 +22,15 @@ type Plain struct {
 
 func (p *Plain) Check(c server.ClientAuth) bool {
 	opts := c.GetOpts()
-	if p.Username != opts.Username || p.Password != opts.Password {
+	if p.Username != opts.Username {
+		return false
+	}
+	// Check to see if the password is a bcrypt hash
+	if isBcrypt(p.Password) {
+		if err := bcrypt.CompareHashAndPassword([]byte(p.Password), []byte(opts.Password)); err != nil {
+			return false
```

Comment:
When authentication fails, the natural path is to assume the password being supplied is wrong, etc.  Especially since you can't _verify_ the password with this tool, so each generation will pick a new salt, so they can't double-check.

Perhaps if `mkpasswd` offered the ability to verify a password against a purported hash, and printed the error, that would be enough.  "How do I test if this password really is correct, since you're saying it's wrong." and it could print the errors.